### ISSUE TITLE
Pylance thinks an attribute is unknown despite "Go to Type Definition" working. Pyright passes. winsdk

### ISSUE THREAD

**Issue by Avasam** [microsoft/pylance-release#5202]
## Environment data

-   Language Server version: Pylance v2023.11.102
-   OS and version: Windows Version	10.0.19045 Build 19045
-   Python version (& distribution if applicable, e.g. Anaconda): Python 3.11.6

## Code Snippet

Here's my code snippet, without my own comments, and I added Pylance reported errors as comments
```python
import asyncio

from winsdk.windows.ai.machinelearning import LearningModelDevice, LearningModelDeviceKind
from winsdk.windows.media.capture import MediaCapture


def get_direct3d_device():  # Return type is unknownPylancereportUnknownParameterType
    media_capture = MediaCapture()

    async def init_mediacapture():
        # "IAsyncAction" is not awaitablePylancereportGeneralTypeIssues
        await (media_capture.initialize_async() or asyncio.sleep(0))

    asyncio.run(init_mediacapture())
    # Type of "direct_3d_device" is "Unknown | None"PylancereportUnknownVariableType
    direct_3d_device = media_capture.media_capture_settings and media_capture.media_capture_settings.direct3_d11_device
    if not direct_3d_device:
        try:
            direct_3d_device = LearningModelDevice(LearningModelDeviceKind.DIRECT_X_HIGH_PERFORMANCE).direct3_d11_device
        except Exception:
            pass
    if not direct_3d_device:
        raise OSError("Unable to initialize a Direct3D Device.")
    return direct_3d_device  # Return type is unknownPylancereportUnknownVariableType
```

## Repro Steps

1. Create a venv
2. `python -m pip install winsdk==1.0.0b10`
3. Configure strict type-checking
4. Paste above code
5. See pylance raising errors about Unkown types. Hovering also shows partial unknowns
6. Run pyright: pass (currently on 1.1.338, but dozens of previous versions also pass)

## Expected behavior

Full type to be known by Pylance, not partially Unknown

Pyright: `information: Type of "direct_3d_device" is "IDirect3DDevice | None"`

## Actual behavior

Partial Unknowns

Pylance:
![image](https://github.com/microsoft/pylance-release/assets/1350584/ea4d8fbb-d0d2-4d1a-a152-1d60229db4a9)

## Logs

[Python Language Server log.txt](https://github.com/microsoft/pylance-release/files/13537632/Python.Language.Server.log.txt)

**Comment by erictraut** [comment]
I'm not able to repro the problem above. I wonder if your Pylance install or your venv are corrupt. Are you able to repro the problem with a fresh project and fresh venv?

**Comment by WorldlyCartos** [comment]
Having same issue with updating python extension yesterday. Switched to the pre-release hoping it would resolve the issue but it hasn't.

**Comment by debonte** [comment]
I'm unable to repro this on Windows with 2023.11.102. I don't get any diagnostics and the type of `direct_3d_device` seems correct:
![image](https://github.com/microsoft/pylance-release/assets/12820898/c9f00887-5d73-4585-8393-e2efe3456aad)

**Comment by Avasam** [comment]
> I'm unable to repro this on Windows with 2023.11.102. I don't get any diagnostics and the type of `direct_3d_device` seems correct: ![image](https://private-user-images.githubusercontent.com/12820898/287574425-c9f00887-5d73-4585-8393-e2efe3456aad.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE3MTAzODAsIm5iZiI6MTcwMTcxMDA4MCwicGF0aCI6Ii8xMjgyMDg5OC8yODc1NzQ0MjUtYzlmMDA4ODctNWQ3My00NTg1LTgzOTMtZTJlZmUzNDU2YWFkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA0VDE3MTQ0MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE4ZmYyYmE3Mzk3NTQzYWQ4NjUyZWJiZjQ1ZGU4OTljNjRiZTkwMzcxMWY0YjNmYWQxZWViYmE0MWYyOTYyOGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.B9uKYXqFI-0aP45CDCv0OQ2sD7AbvGbZW_a0BRN8fAg)

Just wanna make sure, you are in strict mode right? Otherwise the `"IAsyncAction" is not awaitable` issue has been happening to me since quite a few versions back, in basic mode, and in current non-pre-release version (just didn't have time to check if it was a pyright desync back then, and figured it might fix itself with an update by the time I have to work on that project again. But it got worse ^^" )

Another possible variable:
My global default python version is 3.9, my venv is 3.11. In case that makes a difference. (but once again, the `"IAsyncAction" is not awaitable` happenned before.

I might has to start bisecting versions if you can't replicate easily.

**Comment by debonte** [comment]
Yes, I've set `"python.analysis.typeCheckingMode": "strict"`.

I don't see any diagnostics when using the latest Pylance release version (2023.11.10) either.

**Comment by Avasam** [comment]
Using the following minimal repro:
```py
from winsdk.windows.media.capture import MediaCapture
async def foo():
    await MediaCapture().initialize_async() # "IAsyncAction" is not awaitable Pylance[reportGeneralTypeIssues]
```
(because as mentioned previously, the rest of the issues only happen in strict mode on pre-release, so maybe we can focus on this first)

I have bisected this diagnostic to show starting `v2023.10.20`. `v2023.10.10` and a couple previous versions don't show any diagnostic for the above example.

**Comment by Avasam** [comment]
Ok I found the difference in configs! 
It's `useLibraryCodeForTypes = false` that I had configured in my VSCode settings but not in pyright configs.
Starting pyright `1.1.330` I get the same diagnostics. 

I believe I've previously kept this to false in my editor because
1. Way back then the default differed between pyright and pylance (ie before pyright `1.1.303`)
2. I wanna know when I use an untyped library (so that I can add missing annotations myself), but still want type completion

So idk what changed in  pyright `1.1.330`, but I guess that for use with `py.typed` libraries that use a mix of stubs and regular inline annotations I should update my settings to go back to the default of `useLibraryCodeForTypes` (`true`) ? Still surprised this only affects one of the libraries I use.
`reportMissingTypeStubs = "warning"` is probably actually what I want.
